### ISSUE TITLE
How to reset attributes?

### ISSUE THREAD

**Issue by bfontaine** [intercom/intercom-go#30]
Hello,

If you take a `Company` for example, its `MonthlySpend` field is *not* serialized if it’s `0` because of the [`omitempty` tag](https://github.com/intercom/intercom-go/blob/03e1eb1ee0a727a72d5ce3985fb43166646184e6/company.go#L28):

```go
type Company struct {
	// ...snip...
	MonthlySpend int32 `json:"monthly_spend,omitempty"`
	//                                      ^
	//                                      |
	//                                      here
```

This means saving `Company{ID: "abc", MonthlySpend: 0}` and `Company{ID: "abc"}` is the same. Shouldn’t you use an `*int32` (`int32` pointer) instead, and the same on all other reset-able fields?

**Comment by josler** [comment]
Yes, one of the goals I have for v2 is to move to pointer fields for all attributes, in order that this makes sense.

We do this already for some boolean values - with the helper `intercom.Bool(true)`. I'd like to know if there's a better alternative than pointer fields, but that seems to be the norm.

**Comment by travega** [comment]
V2 is now available in the wild :) Reopen if you're still experiencing this issue :+1:
